virtualbox/resource_vm.go

```diff
@@ -298,9 +297,13 @@ func resourceVMCreate(ctx context.Context, d *schema.ResourceData, meta interfac
 
 		target := filepath.Join(vm.BaseFolder, fileName)
 
-		copyfile := exec.Command("cp", "-a", opticalDiskImage, target)
-		if err := copyfile.Run(); err != nil {
-			return diag.Errorf("failed to clone *.iso and *.dmg to VM folder: %v", err)
+		source, err := os.ReadFile(opticalDiskImage)
+		if err != nil {
+			return diag.Errorf("failed to read source optical disk image: %v", err)
+		}
+
+		if err := os.WriteFile(target, source, 0644); err != nil {
+			return diag.Errorf("failed to write target optical disk image: %v", err)
```

Comment:
This means we are loading the entire file into memory before writing it, which can be a problem on memory constrained systems. I think in this case using [`io.Copy`](https://pkg.go.dev/io#Copy) and opening up the source as a `io.Reader` and the target as `io.Writer` using the `os.File`, might be better, as it will performed a buffered read and write, meaning only small chunk of the file will be in memory.

Reply:
Of course, will do! Thank you for the comment.

Reply:
done - will test the change later on my windows machine where I did the inital work.